L1Trigger/L1THGCal/plugins/concentrator/HGCalConcentratorProcessorSelection.cc

```diff
@@ -66,74 +77,70 @@ void HGCalConcentratorProcessorSelection::run(const edm::Handle<l1t::HGCalTrigge
       thickness = HGCalTriggerTools::kScintillatorPseudoThicknessIndex_;
     }
 
-    if (coarsenTriggerCells_ || (fixedDataSizePerHGCROC_ && thickness > kHighDensityThickness_)) {
+    HGCalTriggerTools::SubDetectorType subdet = HGCalTriggerTools::hgcal_silicon_CEE;
+    if (!triggerTools_.isScintillator(module_trigcell.second.at(0).detId())){
+      if (triggerTools_.isEm(module_trigcell.first) ) subdet = HGCalTriggerTools::hgcal_silicon_CEE;
+      else subdet = HGCalTriggerTools::hgcal_silicon_CEH; 
+    }
+    else subdet = HGCalTriggerTools::hgcal_scintillator;
```

Comment:
Actually, since this is used in several place, it would be even better to create a new function in the trigger tools to retrieve the subdet, to avoid code dupication.